plugins/changedetection/compare.py

```diff
@@ -0,0 +1,139 @@
+import rasterio as rio
+from rasterio import warp, transform
+import numpy as np
+import cv2
+import json
+from geojson import Feature, FeatureCollection, dumps, Polygon
+from rasteralign import align, align_altitudes
+
+KERNEL_10_10 = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (10, 10))
+KERNEL_20_20 = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (20, 20))
+
+def compare(reference_dsm_path, reference_dtm_path, compare_dsm_path, compare_dtm_path, epsg, resolution, display_type, min_height, min_area):
+    # Read DEMs and align them
+    with rio.open(reference_dsm_path) as reference_dsm, \
+         rio.open(reference_dtm_path) as reference_dtm, \
+         rio.open(compare_dsm_path) as compare_dsm, \
+         rio.open(compare_dtm_path) as compare_dtm:
+        reference_dsm, reference_dtm, compare_dsm, compare_dtm = align(reference_dsm, reference_dtm, compare_dsm, compare_dtm, resolution=resolution)
+        reference_dsm, reference_dtm, compare_dsm, compare_dtm = align_altitudes(reference_dsm, reference_dtm, compare_dsm, compare_dtm)
+
+    # Get arrays from DEMs
+    reference_dsm_array = reference_dsm.read(1, masked=True)
+    reference_dtm_array = reference_dtm.read(1, masked=True)
+    compare_dsm_array = compare_dsm.read(1, masked=True)
+    compare_dtm_array = compare_dtm.read(1, masked=True)
+
+    # Calculate CHMs
+    chm_reference = reference_dsm_array - reference_dtm_array
+    chm_compare = compare_dsm_array - compare_dtm_array
+
+    # Calculate diff between CHMs
+    diff = chm_reference - chm_compare
```

Comment:
should be 
diff = abs(chm_reference - chm_compare)
right? with abs, whos reference and whos compare will not be a matter, it will detect any "changes"
unlike now, either higher or either lower, it is weird.

Reply:
On the first version I actually did that, but it didn't work as expected. I realized it was much more useful to just be able to determine the reference and compare tasks, instead of simply calculating the absolute difference. That way, you can determine what data is more important, and it is easier to ignore noise.